core/src/test/RefereeTest.kt

```diff
@@ -0,0 +1,150 @@
+import gamelogic.*
+import gamelogic.map.Continent
+import gamelogic.map.PoliticalMap
+import gamelogic.occupations.CountryOccupations
+import gamelogic.occupations.Occupation
+import kotlin.test.*
+
+class RefereeTest {
+
+    val nico = "Nico"
+    val eric = "Eric"
+
+    val arg = "Argentina"
+    val bra = "Brasil"
+    val chi = "Chile"
+    val kam = "Kamchatka"
+    val jap = "Japón"
+    val vie = "Vietnam"
+
+    val sudamerica = Continent("Sudamérica", setOf(arg, bra))
+    val asia = Continent("Asia", setOf(kam, jap))
+    val sudamericaLarge = Continent("Sudamérica", setOf(arg, bra, chi))
+    val asiaLarge = Continent("Asia", setOf(kam, jap, vie))
+
+    val politicalMap = PoliticalMap.Builder()
+        .addContinent(sudamerica)
+        .addContinent(asia)
+        .addBorder(arg, bra)
+        .addBorder(kam, jap)
+        .build()
+
+    val politicalMapLarge = PoliticalMap.Builder()
+        .addContinent(sudamericaLarge)
+        .addContinent(asiaLarge)
+        .addBorder(arg, bra)
+        .addBorder(arg, chi)
+        .addBorder(kam, jap)
+        .addBorder(kam, vie)
+        .build()
+
+    val goalNico = Goal(listOf(OccupyContinent(sudamerica)))
+    val goalEric = Goal(listOf(OccupyContinent(asia)))
+
+    val occupationsSample = listOf(
+        Occupation(arg, nico, PositiveInt(1)), Occupation(kam, nico, PositiveInt(1)),
+        Occupation(bra, eric, PositiveInt(1)), Occupation(jap, eric, PositiveInt(1))
+    )
+
+    val sampleReferee = Referee(mutableListOf<PlayerInfo>(PlayerInfo(nico, Color.Blue,goalNico), PlayerInfo(eric, Color.Brown, goalEric)),
+        politicalMap,
+        CountryOccupations(occupationsSample)
+    )
+
+
+    @Test
+    fun `Game is not over if nobody achieved goal`() {
+        assertFalse(sampleReferee.gameIsOver())
+        assertTrue (sampleReferee.winners().isEmpty())
+    }
+
+    @Test
+    fun `Nico wins if he plays against eric (and achieve his goal first)`() {
+        val occupationsNico = listOf(
+            Occupation(arg, nico, PositiveInt(1)), Occupation(kam, nico, PositiveInt(1)),
+            Occupation(bra, nico, PositiveInt(1))
+        )
+        val occupationsEric = listOf(Occupation(jap, eric, PositiveInt(1)))
+
+        val referee = Referee(mutableListOf<PlayerInfo>(PlayerInfo(nico, Color.Blue,goalNico), PlayerInfo(eric, Color.Brown, goalEric)),
+            politicalMap,
+            CountryOccupations(occupationsEric.union(occupationsNico))
+        )
+        assertTrue(referee.gameIsOver())
+        assertTrue (referee.winners().contains(nico) and !referee.winners().contains(eric))
+    }
+
+    @Test
+    fun `Turns change according list of playerInfo`() {
```

Comment:
```suggestion
    fun `Turns change according to list of playerInfo`() {
```

Ya me conocés, soy hinchapelotas